core/domain/topic_services_test.py

```diff
@@ -2440,6 +2849,118 @@ def mock_get_current_time_in_millisecs() -> int:
                     ]
                 )
 
+    def test_get_published_story_exploration_ids_ignores_exp_ids_from_unpublished_stories( # pylint: disable=line-too-long
+        self
+    ) -> None:
+        topic_exp_ids = ['exp_1', 'exp_2', 'exp_3', 'exp_4']
+        self._create_linked_explorations(
+            self.TOPIC_ID, self.story_id_1, topic_exp_ids)
+
+        story_exp_ids = topic_services.get_all_published_story_exploration_ids(
+            self.TOPIC_ID)
+
+        self.assertItemsEqual(story_exp_ids, [])
+
+    def test_get_published_story_exploration_ids_from_published_stories_when_serial_chapter_feature_disabled( # pylint: disable=line-too-long
+        self
+    ) -> None:
+        topic_exp_ids = ['exp_1', 'exp_2', 'exp_3', 'exp_4']
+        self._create_linked_explorations(
+            self.TOPIC_ID, self.story_id_1, topic_exp_ids)
+        topic_services.publish_story(
+            self.TOPIC_ID, self.story_id_1, self.user_id_admin)
+        with self.swap_to_always_return(
```

Comment:
Probably want to use the new decorator approach for feature flags, here and elsewhere.